src/hildebrand/internal/response.cljc

```diff
@@ -163,8 +175,13 @@
 
 (defmethod restructure-response* :hildebrand.response/batch-write-item
   [_ {:keys [unprocessed] :as resp}]
-  (or (maybe-unprocessed-error unprocessed)
-      (with-meta {} resp)))
+  (let [result (with-meta
```

Comment:
Well I guess I was basing this thinking off how failures are routine/expected with dynamodb, and you're expected to retry until the items get processed. 

> The individual PutItem and DeleteItem operations specified in BatchWriteItem are atomic; however BatchWriteItem as a whole is not. If any requested operations fail because the table's provisioned throughput is exceeded or an internal processing failure occurs, the failed operations are returned in the UnprocessedItems response parameter. You can investigate and optionally resend the requests. Typically, you would call BatchWriteItem in a loop. Each iteration would check for unprocessed items and submit a new BatchWriteItem request with those unprocessed items until all items have been processed.

It would also be useful in the channeled version of this to return the unprocessed items back on a channel to make retrying easier versus treating it as an error. Do you think it should be handled differently here?